File: MergeTwoSortedList/mergeTwoSortedList.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} list1
 * @param {ListNode} list2
 * @return {ListNode}
 */
class ListNode {
  constructor(value, next) {
    this.value = (value === undefined ? 0 : value);
    this.next = (next===undefined ? null : next);
  }
}

var mergeTwoLists = function(list1, list2) {
  
  let mergedList = new ListNode();
  let helper = mergedList;
  let finalAnswer = [];

  while (list1 !== null && list2 !== null) {
    if (list1.value < list2.value) {
      helper.next = new ListNode(list1.value);
      list1 = list1.next;
    } else {
      helper.next = new ListNode(list2.value);
      list2 = list2.next;
    }
    helper = helper.next;
  }

  while (mergedList !== null) {
    finalAnswer.push(mergedList.value);
    mergedList = mergedList.next;
  }

  return finalAnswer;
};


let list1 = new ListNode(1, new ListNode(2, new ListNode(4)));
let list2 = new ListNode(1, new ListNode(3, new ListNode(4)));

console.log(mergeTwoLists(list1, list2));

File: MergeTwoSortedList/mergeTwoSortedList.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def mergeTwoLists(self, list1, list2):
        mergeList = ListNode()
        sortingList = mergeList
        while list1 and list2:
            if list1.val > list2.val:
                sortingList.next = ListNode(list2.val)
                list2 = list2.next
            else:
                sortingList.next = ListNode(list1.val)
                list1 = list1.next
            sortingList = sortingList.next
        sortingList.next = list1 if list1 else list2

        return mergeList.next


list1 = ListNode(1, ListNode(2, ListNode(4)))
list2 = ListNode(1, ListNode(3, ListNode(4)))
sol = Solution()
sol.mergeTwoLists(list1, list2)

def printSolution(list):
    while list:
        print(list.val, end="->")
        list = list.next
    print("None")

printSolution(sol.mergeTwoLists(list1, list2))